it_hardware/demo/purchase_order_confirm.xml

```diff
@@ -74,4 +74,40 @@
         <value model="stock.picking" eval="obj().env.ref('it_hardware.purchase_order_3').picking_ids.ids"/>
     </function>
     <function model="purchase.order" name="action_create_invoice" eval="[ref('purchase_order_3')]"/>
+
+    <function model="purchase.order" name="button_confirm" eval="[ref('purchase_order_4')]"/>
+    <function name="write" model="stock.move.line">
+        <value model="stock.move.line" eval="obj().search([
+        ('product_id', '=', ref('product_product_9')),
+        ('picking_id', '=', obj().env.ref('it_hardware.purchase_order_4').picking_ids.id)])[0].id"/>
```

Comment:
(nitpicking) Not a big fan of [0], [1], ... Couldn't you add a condition in the domain that there is no linked lot and add a limit=1 in the search? What do you think?